modules/nextflow/src/main/groovy/nextflow/config/ConfigBuilder.groovy

```diff
@@ -735,5 +735,21 @@ class ConfigBuilder {
         buildConfigObject().toMap()
     }
 
+    /**
+     * Merge two maps recursively avoiding keys to be overwritten
+     *
+     * @param lmap
+     * @param rmap
+     * @return a map resulting of merging lmap and rmap
+     */
+    def mergeMaps(Map lmap, Map rmap) {
+        if (lmap.size() == 0) {
+            return rmap
+        }
+        rmap.each { k, v ->
```

Comment:
What is `rmap` is empty?

Reply:
Shouldn't happen given [this](https://github.com/JoseEspinosa/nextflow/blob/21bf31829ebe461eae3e31a8af5df668e9b07b75/modules/nextflow/src/main/groovy/nextflow/config/ConfigBuilder.groovy#L647-L648) condition that implies that the right map will always hold something. Anyhow, I updated the function on the new commit ad3e173